tests/functional/singleton/vadd.h

```diff
@@ -0,0 +1,8 @@
+// Copyright (c) 2024 RapidStream Design Automation, Inc. and contributors.
+// All rights reserved. The contributor(s) of this file has/have agreed to the
+// RapidStream Contributor License Agreement.
+
+#include <tapa.h>
+
+// when array size is 1, it forms a singleton array with no number in naming
+constexpr int M = 1;
```

Comment:
Global variables defined in a header should be marked as `inline` (https://abseil.io/tips/168).